### PR TITLE
Feature/annotation control

### PR DESCRIPTION
**Pull request by rpspringuel**
This addresses #768 and fixes a bug discovered while working on it.

### CODE REVIEW

File: doc/Command_Index_User.tex [hunk 1]
```diff
@@ -1224,10 +1242,12 @@ \subsection{Distances}\label{distances}
 
 \begin{gdimension}{annotationseparation}
 This space is the one between lines in the annotation (text above the initial).
+
+\textbf{Nota Bene:} This is the absolute space.  If the lower line contains only short letters then it will get moved up so only this space shows (not the space above the letters on a normal line plus this space).  You should use struts to control the line height of the lower line if this is a problem.
 \end{gdimension}
 
 \begin{gdimension}{annotationraise}
-Amount to raise (positive) or lower (negative) the annotation from it's normal position (baseline of the first line aligned with the top line of the staff).
+Amount to raise (positive) or lower (negative) the annotation from it's normal position (set with \verb=\gresetannotationby= and \verb=\gresetannotationvalign=).
```
Comment (henryso) [inline doc/Command_Index_User.tex]: `it's` should be `its`

File: doc/Command_Index_internal.tex [hunk 2]
```diff
@@ -1227,6 +1227,12 @@ \subsection{Flags}
 indicates if the distance should scale when the staff size changes (true)
 or not (false).
 
+\macroname{\textbackslash ifgre@annotationbottomline}{}{gregoriotex-main.tex}
+Boolean used to indicate if the bottom line of the annotation should be used as the control line for it’s initial vertical alignment.
```
Comment (henryso) [inline doc/Command_Index_internal.tex]: `it's` should be `its`

File: tex/gregoriotex-main.tex [hunk 3]
```diff
@@ -537,17 +566,45 @@
...
+    \or %case 2 = bottom of first line
+      \global\gre@dimen@annotationtrueraise=\dp\gre@box@annotation\relax%
+      \gre@debugmsg{annotation}{Aligning to bottom of first line: \the\gre@dimen@annotationtrueraise}%
+    \else% Bug
+      \gre@bug{Invalid value for \protect\gre@count@annotationvalign}%
+    \fi%
   \else%
     \gre@debugmsg{annotation}{We're not on the first line of the annotation.}%
-    \setbox\gre@box@annotation = \hbox to \gre@dimen@temp@three{\vtop spread \gre@dimen@annotationseparation{{\offinterlineskip\noindent\box\gre@box@annotation@old\vss\noindent\box\gre@box@annotation@add}}}%
+    \setbox\gre@box@annotation@old = \hbox to \gre@dimen@temp@three{\gre@leftfill\box\gre@box@annotation\gre@rightfill}
```
Comment (henryso) [inline tex/gregoriotex-main.tex]: Should there be a `%` at the end of this line?

**Comment by rpspringuel** [comment]
Tests are still coming.  I discovered a bug when preparing that PR.

**Comment by rpspringuel** [comment]
Fixes are in and tests are ready.

**Comment by henryso** [comment]
Looks fine to me.

**Pull request merged**